tests/MSBuildDeviceIntegration/Tests/DebuggingTest.cs

```diff
@@ -39,6 +39,7 @@ public void ApplicationRunsWithoutDebugger ([Values (false, true)] bool isReleas
 
 			var proj = new XamarinFormsAndroidApplicationProject () {
 				IsRelease = isRelease,
+				PackageName = GeneratePackageName (),
```

Comment:
Given how many places this is set, shouldn't we just change the *default* value?

https://github.com/xamarin/xamarin-android/blob/214dd76d9441c9974775f0b29feec0d5d8d1097f/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Android/XamarinAndroidApplicationProject.cs#L70

Aside: ^^ is wrong, in that `PackageName` *cannot* (reasonably) be set before the `XamarinAndroidApplicationProject` constructor executes (short of using Reflection to construct an uninitialized object, set the property, and the invoke the constructor…), and thus it'll always hit the `string.Format ("{0}.{0}", ProjectName)` expression.

Couldn't this instead just become:

```csharp
PackageName = GeneratePackageName ();
```

Is there really a problem if *every* project gets this new behavior?

Reply:
The main reason I wanted to avoid this was the new method has a dependency on Unit. `Xamarin.ProjectTools` currently has not such dependency. I didn't want to add it unless we had a really good reason too.

Reply:
I'm rewording the code